File: apps/apple-watch/IslamicDayDialWatch/Geometry.swift
import SwiftUI

/// Polar to Cartesian. 0° = top, clockwise.
/// Mirrors apps/web-dashboard/src/lib/geometry.ts
func polarToXY(cx: Double, cy: Double, r: Double, angleDeg: Double) -> CGPoint {
    let rad = (angleDeg - 90) * .pi / 180
    return CGPoint(
        x: cx + r * cos(rad),
        y: cy + r * sin(rad)
    )
}

/// Arc path for segment. 0° = top, clockwise (matches web describeArc).
func arcPath(cx: Double, cy: Double, r: Double, startDeg: Double, endDeg: Double) -> Path {
    let span = endDeg - startDeg
    guard span > 0 else { return Path() }
    
    let start = polarToXY(cx: cx, cy: cy, r: r, angleDeg: startDeg)
    let end = polarToXY(cx: cx, cy: cy, r: r, angleDeg: endDeg)
    
    var path = Path()
    path.move(to: start)
    // SwiftUI: 0° = right, 90° = bottom. Our 0° = top = -90° in standard.
    path.addArc(
        center: CGPoint(x: cx, y: cy),
        radius: r,
        startAngle: .degrees(startDeg - 90),
        endAngle: .degrees(endDeg - 90),
        clockwise: true
    )
    return path
}

File: apps/web-dashboard/src/lib/geometry.ts
/**
 * Convert angle (0° = top, clockwise) to SVG coordinates.
 */
export function polarToXY(
  cx: number,
  cy: number,
  r: number,
  angleDeg: number,
): { x: number; y: number } {
  const rad = ((angleDeg - 90) * Math.PI) / 180;
  return {
    x: cx + r * Math.cos(rad),
    y: cy + r * Math.sin(rad),
  };
}


/**
 * SVG arc path for a segment of a circle (used with stroke, no fill).
 */
export function describeArc(
  cx: number,
  cy: number,
  r: number,
  startDeg: number,
  endDeg: number,
): string {
  const span = endDeg - startDeg;
  if (span <= 0) return '';
  if (span >= 360) {
    const top = polarToXY(cx, cy, r, 0);
    const bottom = polarToXY(cx, cy, r, 180);
    return (
      `M ${top.x} ${top.y} ` +
      `A ${r} ${r} 0 1 1 ${bottom.x} ${bottom.y} ` +
      `A ${r} ${r} 0 1 1 ${top.x} ${top.y}`
    );
  }
  const start = polarToXY(cx, cy, r, startDeg);
  const end = polarToXY(cx, cy, r, endDeg);
  const largeArc = span > 180 ? 1 : 0;
  return `M ${start.x} ${start.y} A ${r} ${r} 0 ${largeArc} 1 ${end.x} ${end.y}`;
}
